experiments/modern/src/utils.js

```diff
@@ -139,3 +140,44 @@ export function findDescendantByTypeAndId(node, type, id) {
 
   return null;
 }
+
+function getPreviousSiblings(node, parent) {
+  const children = parent.children;
+  for (let i = 0; i < children.length; i++) {
+    if (children[i] === node) {
+      if (i === 0) {
+        return [];
+      }
+
+      return children.slice(0, i);
+    }
+  }
+
+  return [];
+}
+
+function findDirectDescendantById(node, id) {
+  const children = node.children;
+  for (let i = 0; i < children.length; i++) {
+    const child = children[i];
+    if (child.xmlNode.attributes.id === id) {
+      return child;
+    }
+  }
+}
+
+export function findElementById(node, id) {
+  let currentNode = node;
+  while (currentNode.parent !== null) {
+    let parent = currentNode.parent;
+    const prevSiblings = getPreviousSiblings(currentNode, parent);
```

Comment:
Since we don’t always need to get all the previous siblings, I wonder if we could either in-line the getPreviousSiblings code, or maybe have getPreviousSiblings return an iterable.

Reply:
yea, I can just inline it into a condition that breaks out of the `for` loop.

Thanks!